let ValidationError,CustomErrorHandler;_a15‍.x([["default",()=>_a15‍.o]]);_a15‍.w("joi",[["ValidationError",["ValidationError"],function(v){ValidationError=v}]]);_a15‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);

const errorHandler=(err,req,res,next)=>{
    let statusCode=500;
    let data={
        message:"Internal server error",
        originalError:err.message
    }
    if(err instanceof ValidationError){
        statusCode=422;
        data={
            message:err.message
        }
    }
    if(err instanceof CustomErrorHandler){
        statusCode=err.status;
        data={
            message:err.message
        }
    }
    return res.status(statusCode).json(data);
}
_a15‍.d(errorHandler);